docs/code/getting-started.md

```diff
@@ -319,152 +318,172 @@ Now that we've got all these swanky files laying around, we need a way to tell t
-
-        # where we install data directory (if we have any)
-        set (DATADIR "${CMAKE_INSTALL_PREFIX}/share")
-
-        # what to call that directory where we install data too
-        set (PKGDATADIR "${DATADIR}/com.github.yourusername.yourrepositoryname")
-
-        set (EXEC_NAME "com.github.yourusername.yourrepositoryname")
-        set (RELEASE_NAME "A hello world.")
-        set (VERSION "0.1")
-        set (VERSION_INFO "whats up world")
-
-        # we're about to use pkgconfig to make sure dependencies are installed so let's find pkgconfig first
-        find_package(PkgConfig)
+The next thing we need is a build system. The build system that we're going to be using is called [Meson](http://mesonbuild.com/). We already installed the `meson` program at the beginning of this book when we installed `elementary-sdk`. What we're going to do in this step is create the files that tell Meson how to install your program. This includes all the rules for building your source code as well as correctly installing your .desktop and appdata files and the binary app that results from the build process.
 
-        # now let's actually check for the required dependencies
-        pkg_check_modules(DEPS REQUIRED gtk+-3.0)
+Create a new file in your project's root folder called "meson.build". We've included some comments along the way to explain what each section does. You don't have to copy those, but type the rest into that file:
 
-        add_definitions(${DEPS_CFLAGS})
-        link_libraries(${DEPS_LIBRARIES})
-        link_directories(${DEPS_LIBRARY_DIRS})
+        # project name and programming language
+        project('com.github.yourusername.yourrepositoryname', 'vala', 'c')
 
-        # make sure we have vala
-        find_package(Vala REQUIRED)
-        # make sure we use vala
-        include(ValaVersion)
-        # make sure it's the desired version of vala
-        ensure_vala_version("0.16" MINIMUM)
-
-        # files we want to compile
-        include(ValaPrecompile)
-        vala_precompile(VALA_C ${EXEC_NAME}
-            src/hello-again.vala
-
-        # tell what libraries to use when compiling
-        PACKAGES
-            gtk+-3.0
+        # Create a new executable, list the files we want to compile, list the dependencies we need, and install
+        executable(
+            meson.project_name(),
+            'src/Application.vala',
+            dependencies: [
+                dependency('gtk+-3.0')
+            ],
+            install: true
         )
 
-        # tell cmake what to call the executable we just made
-        add_executable(${EXEC_NAME} ${VALA_C})
-
-        # install the binaries we just made
-        install (TARGETS ${EXEC_NAME} RUNTIME DESTINATION bin)
+        #Install our .desktop file so the Applications Menu will see it
+        install_data(
+            join_paths('data', meson.project_name() + '.desktop'),
+            install_dir : join_paths(get_option('datadir'), 'applications')
```

Comment:
I think this has an extra space and should be changed to `install_dir: join_paths(get_option('datadir'), 'applications')`.